File: src/ch/d/peters/swisstraintimetable/interfaces/IDataHandler.java
package ch.d.peters.swisstraintimetable.interfaces;

import java.util.List;

/**
 * Data
 *
 * @author Daniel Peters
 * @version 1.0
 */
public interface IDataHandler<T> {
  List<T> get();

  void put(List<T> element);

  void put(T element);
}

File: src/ch/d/peters/swisstraintimetable/interfaces/IDataHandler.kt
package ch.d.peters.swisstraintimetable.interfaces

/**
 * Data handler interface.
 *
 * @author Daniel Peters
 * @version 1.0
 */
interface IDataHandler<T> {
  fun get(): List<T>

  fun put(elements: List<T>)

  fun put(element: T)
}
